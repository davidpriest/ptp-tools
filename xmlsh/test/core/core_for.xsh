# core_for.xsh
# tests of for

for a in a b c ; do
	echo $a ;
done

for a in a b c ; do echo $a ; done

# test leftover variable
echo After for a is $a

set xa xb xc
for a ; do echo $a ; done



# 2 level for
for a in a b c ; do
   for  b in c d e ; do
      echo -n "$a$b "
   done
   echo
done

echo 1 level break
for a in a b c ; do
 [ $a = "b" ] && break ;
 echo a is $a
 [ $a != "a" ] && echo Failed  break
done

echo 1 level continue
for a in a b c ; do
 [ $a = "b" ] && continue ;
 [ $a = "b" ] && echo Failed continue 
 echo a is $a
done


echo 2 level for with break 1
for a in a b c ; do
   for  b in c d e ; do
       echo -n "$a$b "
       [ $b = "d" ] && break 1;
       [ $b = "d" ] && echo Failed level 1 break
   done
   echo
done



echo 2 level for with break 2
for a in a b c ; do
   for  b in c d e ; do
       echo -n "$a$b "
       [ $b = "d" ] && break 2;
       [ $b = "d" ] && echo Failed level 1 break
   done
   echo Failed level 2 break 
done
echo



echo 2 level for with continue 1
for a in a b c ; do
   for  b in c d e ; do
       echo -n "$a$b "
       [ $b = "d" ] && continue 1;
       [ $b = "d" ] && echo Failed level 1 continue
   done
   echo
done



echo 2 level for with continue 2
for a in a b c ; do
   for  b in c d e ; do
       echo -n "$a$b "
       [ $b = "d" ] && continue 2;
       [ $b = "d" ] && echo Failed level 1 continue
   done
   echo Failed continue
done
echo
# test IO redirection

[ -d $TMPDIR/_xmlsh ] || mkdir $TMPDIR/_xmlsh
T1=$TMPDIR/_xmlsh/for1.tmp
T2=$TMPDIR/_xmlsh/for2.tmp

echo line1 > $T1
echo line2 >> $T1
echo line3 >> $T1

for a in a b c  ; do
   read line 
   echo $line
done < $T1 > $T2

echo Output is 
cat $T2

# For loop with $() 
for a in $(echo a1; echo b1 ; echo c1  ) ; do
   echo arg $a
done

rm $T1 $T2 
