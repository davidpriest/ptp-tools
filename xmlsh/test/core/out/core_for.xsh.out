a
b
c
a
b
c
After for a is c
xa
xb
xc
ac ad ae 
bc bd be 
cc cd ce 
1 level break
a is a
1 level continue
a is a
a is c
2 level for with break 1
ac ad 
bc bd 
cc cd 
2 level for with break 2
ac ad 
2 level for with continue 1
ac ad ae 
bc bd be 
cc cd ce 
2 level for with continue 2
ac ad bc bd cc cd 
Output is
line1
line2
line3
arg a1
arg b1
arg c1
